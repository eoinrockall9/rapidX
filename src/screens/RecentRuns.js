import React, { Component } from 'react';
import { View, Text, StyleSheet, TextInput, Button, Dimensions, ScrollView } from 'react-native';
import { List, ListItem, Card, CardItem, Body } from 'native-base'
import ItemComponent from '../components/ItemComponent';
import PropTypes from 'prop-types';

import { db } from '../config';

screenHeight = Math.round(Dimensions.get('window').height);

export default class RecentRuns extends Component {
  state = {
    items: [],
    string: '',
    itemsRef: '/runs/anyDistance/',
    distance: '',
    visible: true, 
    itemsLength: 0
  };

  noVisibility = () => {
    this.setState({ visible: false })
  }

  yesVisibility = () => {
    this.setState({ visible: true })
  }

  static propTypes = {
    items: PropTypes.array.isRequired
  };

  handleDistance = e => {
    this.setState({
      distance: e.nativeEvent.text
    });
    
  };

  onUpdate = () => {
    this.setState({
      itemsRef: '/runs/' + this.state.distance
    })
    this.noVisibility();
    this.componentDidMount();
  }

  onShowAll = () => {
    this.setState({
      itemsRef: '/runs/anyDistance'
    })
    this.yesVisibility();
    this.componentDidMount();
  }

  componentDidMount() {

    console.log(this.state.itemsRef)

    db.ref(this.state.itemsRef).on('value', snapshot => {
      //console.log("Snap" + JSON.stringify(snapshot.val()));
      let i = 0
      let items = []
      let length = 0

      snapshot.forEach((subSnapshot) => {
        // // var key = Object.keys(snapshot.val())[index]
        // //console.log("Subsnap" + JSON.stringify(subSnapshot.val()))
        // let data = subSnapshot.val()
        // let oneItems = Object.values(data);
        // items.push(oneItems)
        subSnapshot.forEach((superSubSnapshot) => {
          let data = superSubSnapshot.val()
          let oneItems = Object.values(data);
          items.push(oneItems)
        });

      });
      console.log(items)
      length = items.length
      this.setState({items : items})
      this.setState({ itemsLength: length})
  
    });
  }

  render() {

      return (
      <React.Fragment>
        <View>
          <TextInput onChange={this.handleDistance} style={{ height: 40, borderColor: 'gray', borderWidth: 1 }} placeholder="Enter Distance"/>
          <Button title="Update" onPress={this.onUpdate}/>
          <Button title="See all runs" onPress={this.onShowAll}/>
        </View>
        <ScrollView>
        <View style={{opacity: this.state.visible ? 100 : 0, height: this.state.visible ? screenHeight : 0}}>       
          <Card style={{height: this.state.items[this.state.itemsLength-2] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>{this.state.items[this.state.itemsLength-2]}k - {this.state.items[this.state.itemsLength-1]}</Text>
              </Body>
            </CardItem>
          </Card>
          
          <Card style={{height: this.state.items[this.state.itemsLength-2] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>{this.state.items[this.state.itemsLength-4]}k - {this.state.items[this.state.itemsLength-3]}</Text>
              </Body>
            </CardItem>
          </Card>

          <Card style={{height: this.state.items[4] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-6]}k - {this.state.items[this.state.itemsLength-5]}</Text>
              </Body>
            </CardItem>
          </Card>
          
          <Card style={{height: this.state.items[6] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-8]}k - {this.state.items[this.state.itemsLength-7]}</Text>
              </Body>
            </CardItem>
          </Card>

          <Card style={{height: this.state.items[8] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-10]}k - {this.state.items[this.state.itemsLength-9]}</Text>
              </Body>
            </CardItem>
          </Card>
          
          <Card style={{height: this.state.items[10] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-12]}k - {this.state.items[this.state.itemsLength-11]}</Text>
              </Body>
            </CardItem>
          </Card>

          <Card style={{height: this.state.items[12] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-14]}k - {this.state.items[this.state.itemsLength-13]}</Text>
              </Body>
            </CardItem>
          </Card>
          
          <Card style={{height: this.state.items[14] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-16]}k - {this.state.items[this.state.itemsLength-15]}</Text>
              </Body>
            </CardItem>
          </Card>

          <Card style={{height: this.state.items[16] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-18]}k - {this.state.items[this.state.itemsLength-17]}</Text>
              </Body>
            </CardItem>
          </Card>
          
          <Card style={{height: this.state.items[18] == null ? 0 : 50}}>
            <CardItem>
              <Body>
              <Text>{this.state.items[this.state.itemsLength-20]}k - {this.state.items[this.state.itemsLength-19]}</Text>
              </Body>
            </CardItem>
          </Card>
        </View>
        </ScrollView>

        <View style={{opacity: this.state.visible ? 0 : 100, height: this.state.visible ? 0 : screenHeight}}>       
        <Card style={{height: this.state.items[0] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>
                  {this.state.items[this.state.items.length-1]}
                </Text>
              </Body>
            </CardItem>
          </Card>
          <Card style={{height: this.state.items[1] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>
                  {this.state.items[this.state.items.length-2]}
                </Text>
              </Body>
            </CardItem>
          </Card>
          <Card style={{height: this.state.items[2] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>
                  {this.state.items[this.state.items.length-3]}
                </Text>
              </Body>
            </CardItem>
          </Card>
          <Card style={{height: this.state.items[3] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>
                  {this.state.items[this.state.items.length-4]}
                </Text>
              </Body>
            </CardItem>
          </Card>
          <Card style={{height: this.state.items[4] == null ? 0 : 50}}>
            <CardItem>
              <Body>
                <Text>
                  {this.state.items[this.state.items.length-5]}
                </Text>
              </Body>
            </CardItem>
          </Card>
        </View>
      </React.Fragment>
      );
    
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#ebebeb'
  }
});