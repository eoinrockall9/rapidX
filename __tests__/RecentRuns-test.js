import React from 'react';
import RecentRuns from '../src/screens/RecentRuns';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<RecentRuns />).toJSON();
    expect(tree).toMatchSnapshot();
})