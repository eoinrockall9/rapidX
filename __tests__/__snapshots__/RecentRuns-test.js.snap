// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <View>
    <TextInput
      allowFontScaling={true}
      onChange={[Function]}
      placeholder="Enter Distance"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Update
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          See all runs
        </Text>
      </View>
    </View>
  </View>,
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "height": 1334,
            "opacity": 100,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "height": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
            >
              <Text>
                k - 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "height": 0,
        "opacity": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>,
]
`;
